src/game.cpp

```diff
@@ -8917,13 +8937,29 @@ bool game::walk_move( const tripoint &dest_loc )
     }
 
     if( m.impassable( dest_loc ) && !pushing && !shifting_furniture ) {
+        if( vp_there && u.mounted_creature && u.mounted_creature.get()->has_flag( MF_RIDEABLE_MECH ) &&
```

Comment:
```suggestion
        if( vp_there && u.mounted_creature && u.mounted_creature->has_flag( MF_RIDEABLE_MECH ) &&
```
`mounted_creature` is a `std::shared_ptr`, which has an `operator->` defined so you can use it like a pointer directly.

Applies to all the other places where you call `get()` as well.

Reply:
What im doing currently then is to put it to a local variable when its used more than once after its checked, for better readability.

But if its a one-line thing after is_mounted() then I can use mounted_creature-> directly without setting it to a local variable.

Hope thats suitable?